ppa/archive/static/archive/js/search.js

```diff
@@ -1,63 +1,71 @@
-import DateHistogram from './histogram';
+import ReactiveForm from './ReactiveForm'
+import Histogram from './Histogram'
 
 $(function(){
 
+    /* components */
+    const archiveSearchForm = new ReactiveForm('.form')
+    const dateHistogram = new Histogram('#histogram')
+
     /* dom */
-    const $$searchForm = $('.form')
-    const $$sortLinks = $('.sort .item')
+    const $$results = $('.ajax-container')
+    const $$paginationTop = $('.pagination').first()
+    const $$resultsCount = $('.workscount .count')
     const $$clearDatesLink = $('.clear-selection')
-    const $$collectionInputs = $('#collections input');
-    const $$histogram = $('#histogram')
+    const $$checkboxes = $('.ui.checkbox')
     const $$minDateInput = $('#id_pub_date_0')
     const $$maxDateInput = $('#id_pub_date_1')
-    const $$minDate = $('.min-date')
-    const $$maxDate = $('.max-date')
+    const $$sortInputs = $('.sort input')
+    const $$collectionInputs = $('#collections input')
 
+    /* bindings */
+    archiveSearchForm.onStateChange = submitForm
+    $$clearDatesLink.click(onClearDates)
+    $$sortInputs.change(onSortChange)
+    $$collectionInputs.change(onCollectionChange)
+    $$checkboxes.checkbox() // this is just a standard semantic UI behavior
+    onPageLoad() // misc functions that run once on page load
+    
     /* functions */
-    function changeSort(event) {
-        $(event.target).siblings().find('input[checked=""]').removeAttr('checked')
-        $(event.target).find('input').attr('checked', '')
+    function submitForm(state) {        
+        let url = `?${$.param(state)}` // serialize state using $.param to make querystring
+        window.history.pushState(state, 'PPA Archive Search', url) // update the URL bar
+        let req = fetch(`/archive/${url}`, { // create the submission request
+            headers: { // this header is needed to signal ajax request to django
+                'X-Requested-With': 'XMLHttpRequest'
+            }
+        })
+        req.then(res => res.text()).then(html => { // submit the form and get html back
+            $$paginationTop.html($(html).find('.pagination').html()) // update the top pagination
+            dateHistogram.data = JSON.parse($(html).find('pre').html()) // update the histogram
+            $$resultsCount.html($(html).find('.data .results-count').html()) // update the results count
+            $$results.html(html) // update the results
+            document.dispatchEvent(new Event('ZoteroItemUpdated', { // notify Zotero of changed results
```

Comment:
Saw this and remembered to test the Zotero integration - it is updating properly and getting results for current items after the ajax form submission.

Reply:
awesome!!

Reply:
@thatbudakguy should we suggest updated documentation for the Zotero docs? Not sure if they have an easy mechanism for that. Here's where I found the old code I shared with you before: https://www.zotero.org/support/dev/exposing_metadata

Reply:
at least one of the methods in their sample has this deprecation message on MDN:
>This feature has been removed from the Web standards. Though some browsers may still support it, it is in the process of being dropped. Avoid using it and update existing code if possible; see the compatibility table at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

so, probably worth letting them know at least? do you know if we could contact someone, or file a PR? don't know where the docs are hosted or if they're public...